File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.cpp
class Solution {
public:
  int numberOfSteps(int num){
    int steps = 0; //for count steps
    
    while(num>0) {
        if((num & 1) == 0) { //if num is even
            num >>= 1;       //then divide by 2
        } else {
            num--;           //subtract 1
        }
        steps++;             //count total stpes
    }
    return steps;            //return steps
}
};

File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.java
class Solution {
    public int numberOfSteps(int num) {
      int ans=0;
        
        while(num!=0)
        {
            if(num%2==0)
            {
                num=num/2;
                ans=ans+1;
            }
            else{
                num=num-1;
                ans=ans+1;
            }
            
            
            
        }
        
        
        return ans;
        
    }
}